homeassistant/components/rainbird/__init__.py

```diff
@@ -55,15 +55,39 @@
     extra=vol.ALLOW_EXTRA,
 )
 
+ATTR_DAYS = "days"
+SERVICE_DELAY_IRRIGATION = "delay_irrigation"
+SERVICE_SCHEMA_DELAY_IRRIGATION = vol.Schema(
+    {
+        vol.Required(ATTR_DAYS): cv.positive_int,
+    }
+)
+
 
 def setup(hass, config):
     """Set up the Rain Bird component."""
 
     hass.data[DATA_RAINBIRD] = []
     success = False
+
     for controller_config in config[DOMAIN]:
         success = success or _setup_controller(hass, controller_config, config)
 
+    controller = None
+    if success:
+        controller = hass.data[DATA_RAINBIRD][0]
```

Comment:
I would prefer to use _setup_controller instead of use the hass.data dictionary and the nested array....

Reply:
We can only register a service once. We need to address the correct controller somehow. We could pass the host address of the controller we want to target in the service data and look up the controller with that information (`controller.rainbird_client.rainbird_server`).